### PR TITLE
Add API for s3 bucket

### PR DESCRIPTION
**Pull request by Karolk99**
https://github.com/jellyfish-dev/jellyfish/pull/117

### CODE REVIEW

File: test/jellyfish/room_test.exs [hunk 1]
```diff
@@ -156,6 +163,13 @@ defmodule Jellyfish.RoomTest do
       assert %Component{type: Component.HLS, metadata: @hls_metadata} = component
     end
 
+    test "when request is valid with s3 credentials", %{client: client, room_id: room_id} do
+      assert {:ok, component} =
+               Room.add_component(client, room_id, %{@hls_component_opts | s3: @s3})
+
+      assert %Component{type: Component.HLS, metadata: @hls_metadata} = component
+    end
```
Comment (Rados13) [inline test/jellyfish/room_test.exs:171]: Shouldn't we add a test for situations when one of the credentials is missing?

File: lib/jellyfish/component/hls.ex [hunk 2]
```diff
@@ -8,18 +8,27 @@ defmodule Jellyfish.Component.HLS do
 
   @behaviour Jellyfish.Component.Deserializer
 
+  @type credentials :: %{
+          access_key_id: String.t(),
+          secret_access_key: String.t(),
+          region: String.t(),
+          bucket: String.t()
+        }
```
Comment (Rados13) [inline lib/jellyfish/component/hls.ex:16]: What do you think about validating this map before sending a request?

**Review by roznawsk — APPROVED** [review]

File: lib/jellyfish/room.ex [hunk 3]
```diff
@@ -253,6 +257,22 @@ defmodule Jellyfish.Room do
...
   end
 
+  defp validate_component(%RTSP{}), do: :ok
+  defp validate_component(%HLS{s3: nil}), do: :ok
+  defp validate_component(%HLS{s3: %{} = s3}), do: validate_s3_credentials(s3)
+  defp validate_component(_component), do: {:error, :component_validation}
+
+  defp validate_s3_credentials(credentials) do
+    keys = Map.keys(credentials)
+    if @s3_kyes -- keys == [] and keys -- @s3_kyes, do: :ok, else: {:error, :component_validation}
```
Comment (Rados13) [inline lib/jellyfish/room.ex:267]: Isn't this `if` condition missing something?

**Review by Rados13 — APPROVED** [review]

File: lib/jellyfish/room.ex [hunk 4]
```diff
@@ -31,9 +31,12 @@ defmodule Jellyfish.Room do
   """
 
   alias Tesla.Env
+  alias Jellyfish.Component.{HLS, RTSP}
   alias Jellyfish.{Client, Component, Peer, Utils}
   alias Jellyfish.Exception.StructureError
 
+  @s3_kyes [:access_key_id, :secret_access_key, :region, :bucket]
```
Comment (Rados13) [inline lib/jellyfish/room.ex:38]: Shouldn't it be `@s3_keys`?

**Pull request merged**